automation/src/main/java/org/wso2/testgrid/automation/parser/FunctionalTestResultParser.java

```diff
@@ -81,61 +82,69 @@ public FunctionalTestResultParser(TestScenario testScenario, String testLocation
     public void parseResults() throws JTLResultParserException {
         boolean failureMsgElement = false;
         XMLInputFactory factory = XMLInputFactory.newInstance();
-        String scenarioResultFile = ResultParserUtil.getJTLFile(this.testLocation);
+        String[] scenarioResultFiles = ResultParserUtil.getJTLFiles(this.testLocation);
         String testScenarioName = testScenario.getName();
-        try (InputStream inputStream = new FileInputStream(scenarioResultFile)) {
-            if (logger.isDebugEnabled()) {
-                logger.debug("Parsing scenario-results file of the TestScenario : '"
-                        + testScenarioName + "' using the FunctionalTestResultParser");
-            }
 
-            XMLEventReader eventReader = factory.createXMLEventReader(inputStream);
-            TestCase testCase = null;
-            while (eventReader.hasNext()) {
-                XMLEvent event = eventReader.nextEvent();
-                switch (event.getEventType()) {
-                    case XMLStreamConstants.START_ELEMENT:
-                        StartElement startElement = event.asStartElement();
-                        String elementName = startElement.getName().getLocalPart();
-                        if (HTTP_SAMPLE_ELEMENT.equalsIgnoreCase(elementName) ||
-                                SAMPLE_ELEMENT.equalsIgnoreCase(elementName)) {
-                            testCase = this.buildTestCase(startElement);
-                        } else if (FAILURE_MESSAGE_ELEMENT.equalsIgnoreCase(elementName)) {
-                            failureMsgElement = true;
-                        }
-                        break;
-                    case XMLStreamConstants.CHARACTERS:
-                        Characters characters = event.asCharacters();
-                        if (failureMsgElement) {
-                            testCase.setFailureMessage(characters.getData());
-                            failureMsgElement = false;
-                        }
-                        break;
-                    case XMLStreamConstants.END_ELEMENT:
-                        EndElement endElement = event.asEndElement();
-                        String nodeName = endElement.getName().getLocalPart();
-                        if (HTTP_SAMPLE_ELEMENT.equalsIgnoreCase(nodeName) ||
-                                SAMPLE_ELEMENT.equalsIgnoreCase(nodeName)) {
-                            this.testScenario.addTestCase(testCase);
-                        }
-                        break;
-                    default:
-                        break;
+        if (scenarioResultFiles.length == 0) {
+            logger.warn("Unable to locate jtl files for the scenario : " + testScenarioName);
+            return;
+        }
+
+        for (String jtlFile : scenarioResultFiles) {
+            try (InputStream inputStream = new FileInputStream(Paths.get(testLocation, jtlFile).toString())) {
```

Comment:
This is the API to get an InputStream in Java 7+ - `jav.nio.file.Files#newInputStream`

https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#newInputStream-java.nio.file.Path-java.nio.file.OpenOption...-